test/validate/ferc1_test.py

```diff
@@ -160,3 +164,44 @@ def test_unique_rows_ferc1(pudl_out_ferc1, live_dbs, df_name, unique_subset):
         subset=unique_subset,
         df_name=df_name,
     )
+
+
+@pytest.mark.parametrize(
+    "table_name",
+    [  # some sample wide guys
+        "core_ferc1__yearly_sales_by_rate_schedules_sched304",
+        "core_ferc1__yearly_pumped_storage_plants_sched408",
+        "core_ferc1__yearly_steam_plants_sched402",
+        "core_ferc1__yearly_hydroelectric_plants_sched406",
+        "core_ferc1__yearly_transmission_lines_sched422",
+        # some sample guys found to have higher filtering diffs
+        "core_ferc1__yearly_utility_plant_summary_sched200",
+        "core_ferc1__yearly_plant_in_service_sched204",
+        "core_ferc1__yearly_operating_expenses_sched320",
+        "core_ferc1__yearly_income_statements_sched114",
+    ],
+)
+def test_filter_for_freshest_data(
+    ferc1_engine_xbrl, table_name: pd.DataFrame
+) -> pd.DataFrame:
+    """Test if we are unexpectedly replacing records during filter_for_freshest_data."""
+
+    raw_table_names = TABLE_NAME_MAP_FERC1[table_name]["xbrl"]
+    # sometimes there are many raw tables that go into one
+    # core table, but usually its a string.
+    if isinstance(raw_table_names, str):
+        raw_table_names = [raw_table_names]
+    xbrls_with_periods = chain.from_iterable(
+        (f"raw_ferc1_xbrl__{tn}_instant", f"raw_ferc1_xbrl__{tn}_duration")
+        for tn in raw_table_names
+    )
+    for raw_table_name in xbrls_with_periods:
+        logger.info(f"Checking if our filtering methodology works for {raw_table_name}")
+        xbrl_table = defs.load_asset_value(raw_table_name)
+        if not xbrl_table.empty:
+            primary_keys = get_primary_key_raw_xbrl(
+                raw_table_name.removeprefix("raw_ferc1_xbrl__"), "ferc1"
+            )
+            filter_for_freshest_data_xbrl(
```

Comment:
It's a little confusing that the assertions don't live in the test function, but I guess this works! I think someone confused about what this is actually testing would just go into `filter_for_freshest` and immediately see the assertion errors.

Reply:
yea i absolutely agree with this. it just would have been a lot of contrived passing things around and making little functions to get the right inputs in the right place. this feels simpler for now but also i can def imagine wanting to come back and break this up so the __compare_dedupe_methodologies step happens entirely in the test.